transport/src/test/java/io/netty/channel/pool/SimpleChannelPoolTest.java

```diff
@@ -142,4 +149,95 @@ protected boolean offerChannel(Channel ch) {
         channel2.close().sync();
         group.shutdownGracefully();
     }
+
+    /**
+     * Tests that if channel was unhealthy it is not offered back to the pool.
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testUnhealthyChannelIsNotOffered() throws Exception {
+        EventLoopGroup group = new DefaultEventLoopGroup();
+        LocalAddress addr = new LocalAddress(LOCAL_ADDR_ID);
+        Bootstrap cb = new Bootstrap();
+        cb.remoteAddress(addr);
+        cb.group(group)
+          .channel(LocalChannel.class);
+
+        ServerBootstrap sb = new ServerBootstrap();
+        sb.group(group)
+          .channel(LocalServerChannel.class)
+          .childHandler(new ChannelInitializer<LocalChannel>() {
+              @Override
+              public void initChannel(LocalChannel ch) throws Exception {
+                  ch.pipeline().addLast(new ChannelHandlerAdapter());
+              }
+          });
+
+        // Start server
+        Channel sc = sb.bind(addr).syncUninterruptibly().channel();
+        ChannelPoolHandler handler = new CountingChannelPoolHandler();
+        ChannelPool pool = new SimpleChannelPool(cb, handler);
+        Channel channel1 = pool.acquire().syncUninterruptibly().getNow();
+        pool.release(channel1).syncUninterruptibly();
+        Channel channel2 = pool.acquire().syncUninterruptibly().getNow();
+        //first check that when returned healthy then it actually offered back to the pool.
+        assertSame(channel1, channel2);
+
+        expectedException.expect(IllegalStateException.class);
+        expectedException.expectMessage("Channel is unhealthy not offering it back to pool");
```

Comment:
So that I know for sure that expected exception(not any random) had been thrown.